ui/src/app/domains/views/DomainDetails/DomainsDetails.test.tsx

```diff
@@ -0,0 +1,56 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import { MemoryRouter, Route } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import DomainDetails from "./DomainDetails";
+
+import {
+  domain as domainFactory,
+  domainState as domainStateFactory,
+  rootState as rootStateFactory,
+} from "testing/factories";
+
+const mockStore = configureStore();
+
+describe("DomainDetails", () => {
+  it("shows a spinner if domain has not loaded yet", () => {
+    const state = rootStateFactory({
+      domain: domainStateFactory({ items: [] }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/domain/1", key: "testKey" }]}
+        >
+          <Route exact path="/domain/:id" component={() => <DomainDetails />} />
```

Comment:
@bartaz This is a bit of a weird pattern, but in order to get `useParams` to give you what you want in a test, you have to wrap the *actual* component in this `Route` component.

So in this case we're mocking that the url is at `/domain/1`, but since no there is no domain in state with an id of 1, a spinner should be showing